### PR TITLE
Enable .NET 452 - part 01

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #716

## Changes
- Add .NET 452 to OpenTelemetry.Exporter.OpenTelemetryProtocol
- Add .NET 452 to OpenTelemetry.Instrumentation.Dependencies
- Add .NET 452 to OpenTelemetry.Instrumentation.Dependencies.Tests
- Add .NET 452 to OpenTelemetry.Instrumentation.AspNet.Win
- Add .NET 452 to OpenTelemetry.Instrumentation.AspNet.Tests.Win
- Add .NET 452 to OpenTelemetry.Shims.OpenTracing

### Checklist
- [X] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public API reviewed

The PR will automatically request reviews from [code owners](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CODEOWNERS#L15) and trigger CI build and tests.

### CODE REVIEW

**Comment by eddynaka** [comment]
Questions:
- OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests just targets netcoreapp3.1: should we target 452 and 46?
- OpenTelemetry.Instrumentation.Dependencies.Tests just targets netcoreapp3.1: should we target 452 and 46?

File: src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net452;net461</TargetFrameworks>
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj:3]: Just FYI the .NET Framework instrumentation is turned on via `#if NET461` directives so we'll need to update those in order to actually instrument net452. I'm thinking the reflection will still work fine without changes. The .NET version it spawned from targetted net45 so I checked reference source that far back to make my revisions. But good idea to validate it works after the target is updated.
Example:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/5da574c109d1f6787dec08c47d44d7c294c2832c/src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs#L16

Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj:3]: thank you for reviewing. i will check where we have the tag net461 and add 452 and test.

Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj:3]: Should we rename the filename or should we create a new one for net452?

Comment (CodeBlanch) [reply src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj:3]: Maybe rename to end in ".net4x" instead? @SergeyKanzhelev I think you were the one who originally asked for the extension to be used when we were targeting something for an entire file, how do you feel about that?

File: samples/Exporters/AspNet/Global.asax.cs [hunk 2]
```diff
@@ -3,14 +3,12 @@
 using System.Web.Http;
 using System.Web.Mvc;
 using System.Web.Routing;
-using OpenTelemetry.Trace;
 using OpenTelemetry.Trace.Configuration;
 
 namespace OpenTelemetry.Exporter.AspNet
 {
     public class WebApiApplication : HttpApplication
     {
```
Comment (cijothomas) [inline samples/Exporters/AspNet/Global.asax.cs:11]: please try to avoid unrelated changes. While this is good change, its not related to adding net452, so it's best to do separately.

File: src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs [hunk 3]
```diff
@@ -13,7 +13,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 // </copyright>
-#if NET461
+#if NET452 || NET461
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs:16]: Should we define a const like NETFRAMEWORK or NETFULL which will be set for NET452 NET461 and anything like NET47 etc in future?

File: samples/Exporters/AspNet/Global.asax.cs (hunk 2 above)
Comment (eddynaka) [reply samples/Exporters/AspNet/Global.asax.cs:11]: ok. I will revert those changes.

File: src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs (hunk 3 above)
Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs:16]: Well, NETFRAMEWORK is already a constant. I can change in those places. What do you think?

Comment (CodeBlanch) [reply src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs:16]: NETFRAMEWORK target sounds good to me.

File: test/OpenTelemetry.Instrumentation.Dependencies.Tests/OpenTelemetry.Instrumentation.Dependencies.Tests.csproj [hunk 4]
```diff
@@ -20,14 +20,15 @@
...
-    <PackageReference Include="Microsoft.Data.SqlClient" Version="1.1.1" />
+    <PackageReference Include="Microsoft.Data.SqlClient" Version="1.1.1" Condition="'$(TargetFramework)' != 'net452'" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
     <PackageReference Include="Moq" Version="4.11.0" />
     <PackageReference Include="xunit" Version="2.4.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" Condition="'$(TargetFramework)' == 'net461' OR '$(TargetFramework)' == 'net452'" />
```
Comment (CodeBlanch) [inline test/OpenTelemetry.Instrumentation.Dependencies.Tests/OpenTelemetry.Instrumentation.Dependencies.Tests.csproj:31]: Let's not bring in Newtonsoft.Json for net461, it is using System.Text.Json.

Comment (CodeBlanch) [reply test/OpenTelemetry.Instrumentation.Dependencies.Tests/OpenTelemetry.Instrumentation.Dependencies.Tests.csproj:31]: Scratch that, it looks like OpenTelemetry.Instrumentation.Dependencies.Tests.csproj is using both. Fine to keep!

File: test/OpenTelemetry.Instrumentation.Dependencies.Tests/HttpWebRequestActivitySourceTests.net461.cs [hunk 5]
```diff
@@ -177,17 +177,17 @@ public async Task TestBasicReceiveAndResponseEvents(string method, string queryS
             Assert.Equal(1, eventRecords.Records.Count(rec => rec.Key == "Stop"));
 
             // Check to make sure: The first record must be a request, the next record must be a response.
-            (Activity activity, HttpWebRequest startRequest) = AssertFirstEventWasStart(eventRecords);
+            var wrapper = AssertFirstEventWasStart(eventRecords);
```
Comment (CodeBlanch) [inline test/OpenTelemetry.Instrumentation.Dependencies.Tests/HttpWebRequestActivitySourceTests.net461.cs:180]: I think if you add...
`<PackageReference Include="System.ValueTuple" Version="4.5.0" Condition="'$(TargetFramework)' == 'net452'" />`
...to the csproj than the Tuple syntax would have worked.

Comment (eddynaka) [reply test/OpenTelemetry.Instrumentation.Dependencies.Tests/HttpWebRequestActivitySourceTests.net461.cs:180]: i will change that and undo. thanks for the advice

File: test/OpenTelemetry.Instrumentation.Dependencies.Tests/OpenTelemetry.Instrumentation.Dependencies.Tests.csproj (hunk 4 above)
Comment (CodeBlanch) [reply test/OpenTelemetry.Instrumentation.Dependencies.Tests/OpenTelemetry.Instrumentation.Dependencies.Tests.csproj:31]: Actually, that is not entirely true. It looks like OpenTelemetry.Instrumentation.Dependencies.Tests is using both. But net461 & netcoreapp3.1 targets get their Newtonsoft.Json ref through Microsoft.Data.SqlClient (Microsoft.Data.SqlClient -> Microsoft.IdentityModel.Protocols.OpenIdConnect -> Newtonsoft.Json).

You should be able to do:
`<PackageReference Include="Newtonsoft.Json" Version="12.0.2" Condition="'$(TargetFramework)' == 'net452'" />`

**Comment by cijothomas** [comment]
@eddynaka HttpWebRequestActivitySource.net461.cs  - can you rename these to .netframework as well as a followup PR?

File: src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj (hunk 1 above)
Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetry.Instrumentation.Dependencies.csproj:3]: we will update from net461 to netframework as cijo suggested
